xbmc/filesystem/FileDirectoryFactory.cpp

```diff
@@ -64,20 +64,22 @@ IFileDirectory* CFileDirectoryFactory::Create(const CURL& url, CFileItem* pItem,
 
   std::string strExtension=URIUtils::GetExtension(url);
   StringUtils::ToLower(strExtension);
-  VECADDONS codecs;
-  CAddonMgr::GetInstance().GetAddons(codecs, ADDON_AUDIODECODER);
-  for (size_t i=0;i<codecs.size();++i)
+  if (!strExtension.empty())
```

Comment:
See: https://github.com/xbmc/xbmc/blob/Jarvis/xbmc/filesystem/FileDirectoryFactory.cpp#L72 <- the complete call to GetAddons is useless when it's done for a directory .... it needs > 20 ms ...

If you know something better, please let me know.